packages/seed/src/commands/publish/publishGenerator.ts

```diff
@@ -0,0 +1,181 @@
+import { loggingExeca } from "@fern-api/logging-execa";
+import { TaskContext } from "@fern-api/task-context";
+import { GeneratorReleaseRequest } from "@fern-fern/generators-sdk/api/resources/generators";
+import semver from "semver";
+import { PublishDockerConfiguration } from "../../config/api";
+import { GeneratorWorkspace } from "../../loadGeneratorWorkspaces";
+import { parseGeneratorReleasesFile } from "../../utils/convertVersionsFileToReleases";
+
+interface VersionFilePair {
+    latestVersionFile: string;
+    previousVersionFile: string;
+}
+
+export async function publishGenerator({
+    generator,
+    version,
+    context
+}: {
+    generator: GeneratorWorkspace;
+    version: string | VersionFilePair;
+    context: TaskContext;
+}): Promise<void> {
+    const generatorId = generator.workspaceName;
+    context.logger.info(`Publishing generator ${generatorId}@${version}...`);
+
+    let publishVersion: string;
+    if (typeof version !== "string") {
+        // We were given two version files, so we need to compare them to find if any new
+        // versions have been added since the last publish.
+        const maybeNewVersion = await getNewVersion({
+            generatorId,
+            versionFilePair: version,
+            context
+        });
+
+        if (maybeNewVersion == null) {
+            context.failWithoutThrowing(
+                "No version to publish! There must not have been a new version since the last publish."
+            );
+            return;
+        }
```

Comment:
@armandobelardo what if i modify the changelog to yank a version or to update a release note?